app/api/packets/helpers.ts

```diff
@@ -157,12 +227,64 @@ export async function getPacket(txHash: string): Promise<Packet[]> {
     variables: { txHash }
   };
 
+  const ackPacketSearch = {
+    query: `query Packet($txHash:String!){
+              packets(where: {ackPacket: {transactionHash_eq: $txHash}}){
```

Comment:
You can make a template out of the query with `sendPacket`, `recvPacket` and `ackPacket` used as template vars.